plugin/trino-cassandra/src/test/java/io/trino/plugin/cassandra/TestCassandraTable.java

```diff
@@ -0,0 +1,193 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.cassandra;
+
+import com.google.common.collect.ImmutableList;
+import io.trino.testing.sql.SqlExecutor;
+import io.trino.testing.sql.TestTable;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.lang.Math.min;
+import static java.lang.String.format;
+import static java.lang.String.join;
+import static java.util.stream.Collectors.joining;
+
+public class TestCassandraTable
+        extends TestTable
+{
+    private static final int TABLE_NAME_MAXIMUM_LENGTH = 48;
+
+    private final String tableName;
+
+    public TestCassandraTable(
+            SqlExecutor sqlExecutor,
+            CassandraServer server,
+            String keyspace,
+            String namePrefix,
+            List<ColumnDefinition> columnDefinitions)
+    {
+        this(sqlExecutor, server, keyspace, namePrefix, columnDefinitions, Optional.empty(), ImmutableList.of());
+    }
+
+    public TestCassandraTable(
+            SqlExecutor sqlExecutor,
+            CassandraServer server,
+            String keyspace,
+            String namePrefix,
+            List<ColumnDefinition> columnDefinitions,
+            Optional<String> tableOptions)
+    {
+        this(sqlExecutor, server, keyspace, namePrefix, columnDefinitions, tableOptions, ImmutableList.of());
+    }
+
+    public TestCassandraTable(
+            SqlExecutor sqlExecutor,
+            CassandraServer server,
+            String keyspace,
+            String namePrefix,
+            List<ColumnDefinition> columnDefinitions,
+            List<String> rowsToInsert)
+    {
+        this(sqlExecutor, server, keyspace, namePrefix, columnDefinitions, Optional.empty(), rowsToInsert);
+    }
+
+    public TestCassandraTable(
+            SqlExecutor sqlExecutor,
+            CassandraServer server,
+            String keyspace,
+            String namePrefix,
+            List<ColumnDefinition> columnDefinitions,
+            Optional<String> tableOptions,
+            List<String> rowsToInsert)
+    {
+        super(sqlExecutor, keyspace + "." + namePrefix, tableDefinition(columnDefinitions, tableOptions), columnNames(columnDefinitions), rowsToInsert);
+        this.tableName = getName().substring(keyspace.length() + 1);
+
+        try {
+            server.refreshSizeEstimates(keyspace, tableName);
```

Comment:
Do all tests require this (I only see one usage in the existing tests)? If not then let's leave it to the caller to do.
Otherwise add a comment about why this is needed?

Reply:
> Do all tests require this (I only see one usage in the existing tests)?

I've confirmed that `refreshSizeEstimates` is necessary because the `session` used to execute the `CREATE TABLE` and `INSER INTO` statements in `TestCassandraTable` is not the same instance as the `session` used to execute the `SELECT` statements in `DistributedQueryRunner`.

If we don't flush the table to disk, `DistributedQueryRunner` won't be able to find the table created by `TestCassandraTable`.

Reply:
Thanks for explaining. Add a code comment here with this knowledge.